src/utils/proxySet.ts

```diff
@@ -0,0 +1,96 @@
+import { proxy } from 'valtio'
+
+// properies that we don't want to expose to the end-user
+interface InternalProxySet<T> extends Set<T> {
+  data: T[]
+  toJSON: object
+  hasProxy(value: T): boolean
+}
+
+export function proxySet<T>(
+  initialValues: Iterable<T> | null | undefined = []
```

Comment:
can this work?
```suggestion
  initialValues: Iterable<T> = []
```

Reply:
I believe this would prevent `proxySet(null)`, `new Set(null)` is valid. I used the same signature as `Set`.
![Screenshot 2022-01-14 at 07 37 33](https://user-images.githubusercontent.com/5574267/149462758-3a481d02-5a45-44d9-b311-d7b23da833b7.png)

Reply:
Oh, you are right. `null` has to be clearly defined. `undefined` can be omitted, because of the default parameter.

https://tsplay.dev/Ndd9YN
<img width="516" alt="image" src="https://user-images.githubusercontent.com/490574/149507827-61179c6b-4d06-4265-a508-c9012b0724ae.png">